internal/manager/setup.go

```diff
@@ -176,14 +156,17 @@ func setupAdmissionServer(
 		return nil
 	}
 
-	kongclient, err := managerConfig.GetKongClient(ctx)
+	kongclients, err := managerConfig.GetKongClients(ctx)
 	if err != nil {
 		return err
 	}
 	srv, err := admission.MakeTLSServer(ctx, &managerConfig.AdmissionServer, &admission.RequestHandler{
 		Validator: admission.NewKongHTTPValidator(
-			kongclient.Consumers,
-			kongclient.Plugins,
+			// For now using first client is kind of OK. We should take a look at this
+			// sooner rather than later.
+			// https://github.com/Kong/kubernetes-ingress-controller/issues/3363
```

Comment:
Can you expand a bit more on why this is  "kind of ok"? Just a short sentence about where's the problem, as it can be a bit obscure at a first glance.

Reply:
Added a more descriptive comment. I'm not 100% convinced about this as I look at this now since there are 2 actions that are related to kong clients in this admission server:

- validating plugins: this should be OK as in situations where a plugin doesn't (yet) exist in another Kong instance it will get rejected
- [checking if a consumer already exist](https://github.com/Kong/kubernetes-ingress-controller/blob/2ef1a64bc3c8b8cb8b68d97457260e3c9724b052/internal/admission/validator.go#L88-L97): this might be tricky as there might be instances where the consumer exists already in another Kong (not the first one which we have client for) but doesn't exist in the first client (which we have the client for) and as a result of this we allow the creation of this Consumer.

That said: those should be fairly rare and I'm not sure how seriously we should treat those.